pkg/operation/botanist/component/dependencywatchdog/bootstrap.go

```diff
@@ -107,193 +112,22 @@ type bootstrapper struct {
-				Template: corev1.PodTemplateSpec{
-					ObjectMeta: metav1.ObjectMeta{
-						Labels: b.podLabels(),
-					},
-					Spec: corev1.PodSpec{
-						PriorityClassName:             v1beta1constants.PriorityClassNameSeedSystem800,
-						ServiceAccountName:            serviceAccount.Name,
-						TerminationGracePeriodSeconds: pointer.Int64(5),
-						Containers: []corev1.Container{{
-							Name:            name,
-							Image:           b.values.Image,
-							ImagePullPolicy: corev1.PullIfNotPresent,
-							Command:         b.containerCommand(),
-							Ports: []corev1.ContainerPort{{
-								Name:          "metrics",
-								ContainerPort: 9643,
-								Protocol:      corev1.ProtocolTCP,
-							}},
-							Resources: corev1.ResourceRequirements{
-								Requests: corev1.ResourceList{
-									corev1.ResourceCPU:    resource.MustParse("200m"),
-									corev1.ResourceMemory: resource.MustParse("256Mi"),
-								},
-								Limits: corev1.ResourceList{
-									corev1.ResourceMemory: resource.MustParse("512Mi"),
-								},
-							},
-							VolumeMounts: []corev1.VolumeMount{{
-								Name:      volumeName,
-								MountPath: volumeMountPath,
-								ReadOnly:  true,
-							}},
-						}},
-						Volumes: []corev1.Volume{{
-							Name: volumeName,
-							VolumeSource: corev1.VolumeSource{
-								ConfigMap: &corev1.ConfigMapVolumeSource{
-									LocalObjectReference: corev1.LocalObjectReference{
-										Name: configMap.Name,
-									},
-								},
-							},
-						}},
-					},
-				},
-			},
-		}
-
-		updateMode = vpaautoscalingv1.UpdateModeAuto
-		vpa        = &vpaautoscalingv1.VerticalPodAutoscaler{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      b.name() + "-vpa",
-				Namespace: b.namespace,
-			},
-			Spec: vpaautoscalingv1.VerticalPodAutoscalerSpec{
-				TargetRef: &autoscalingv1.CrossVersionObjectReference{
-					APIVersion: appsv1.SchemeGroupVersion.String(),
-					Kind:       "Deployment",
-					Name:       deployment.Name,
-				},
-				UpdatePolicy: &vpaautoscalingv1.PodUpdatePolicy{
-					UpdateMode: &updateMode,
-				},
-				ResourcePolicy: &vpaautoscalingv1.PodResourcePolicy{
-					ContainerPolicies: []vpaautoscalingv1.ContainerResourcePolicy{{
-						ContainerName: vpaautoscalingv1.DefaultContainerResourcePolicy,
-						MinAllowed: corev1.ResourceList{
-							corev1.ResourceCPU:    resource.MustParse("25m"),
-							corev1.ResourceMemory: resource.MustParse(vpaMinAllowedMemory),
-						},
-					}},
-				},
-			},
-		}
-
+		registry            = managedresources.NewRegistry(kubernetes.SeedScheme, kubernetes.SeedCodec, kubernetes.SeedSerializer)
 		podDisruptionBudget client.Object
 		maxUnavailable      = intstr.FromInt(1)
 	)
 
+	configMap, err := b.getConfigMap()
+	if err != nil {
+		return err
+	}
+	serviceAccount := b.getServiceAccount()
+	saNamespacedName := types.NamespacedName{Namespace: serviceAccount.Namespace, Name: serviceAccount.Name}
```

Comment:
Why do we need this variable? You could directly pass the name and namespace.

Reply:
Yes that is fine. We can change the signature of `getClusterRoleAndClusterRoleBinding` and `getRoleAndRoleBinding` to accept a `namespace` and `name` instead of `NamespacedName`.